apis/projectcontour/v1/httpproxy.go

```diff
@@ -101,17 +101,120 @@ type HeaderMatchCondition struct {
 	NotExact string `json:"notexact,omitempty"`
 }
 
+// ExtensionServiceReference names an ExtensionService resource.
+type ExtensionServiceReference struct {
+	// API version of the referent.
+	// If this field is not specified, the default "projectcontour.io/v1alpha1" will be used
+	//
+	// +optional
+	// +kubebuilder:validation:MinLength=1
+	APIVersion string `json:"apiVersion,omitempty" protobuf:"bytes,5,opt,name=apiVersion"`
+
+	// Kind of the referent.
+	// If this field is not specified, the default "ExtensionService" will be used.
+	//
+	// More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#types-kinds
+	//
+	// +optional
+	// +kubebuilder:validation:MinLength=1
+	Kind string `json:"kind,omitempty" protobuf:"bytes,1,opt,name=kind"`
+
+	// Namespace of the referent.
+	// If this field is not specifies, the namespace of the resource that targets the referent will be used.
+	//
+	// More info: https://kubernetes.io/docs/concepts/overview/working-with-objects/namespaces/
+	//
+	// +optional
+	// +kubebuilder:validation:MinLength=1
+	Namespace string `json:"namespace,omitempty" protobuf:"bytes,2,opt,name=namespace"`
+
+	// Name of the referent.
+	//
+	// More info: https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#names
+	//
+	// +required
+	// +kubebuilder:validation:MinLength=1
+	Name string `json:"name,omitempty" protobuf:"bytes,3,opt,name=name"`
+}
+
+// AuthorizationServer configures an external server to authenticate
+// client requests. The external server must implement the Envoy
+// external authorization GRPC protocol. Currently, the
+// [v2](https://www.envoyproxy.io/docs/envoy/latest/api-v2/service/auth/v2/external_auth.proto)
+// protocol is always used, but authorization server authors should implement
+// the v3 protocol as well in the expectation that it will be supported
+// in future.
+type AuthorizationServer struct {
+	// ServiceRef specifies the extension resource that will authorize client requests.
+	//
+	// +required
+	ServiceRef ExtensionServiceReference `json:"serviceRef"`
```

Comment:
There's only one kind of ref, so I didn't feel like this needed disambiguations. The YAML for `serviceRef` seems pretty readable and evocative?

Reply:
What's ambiguous is to a user the api states it's a `service` `ref` which could be an extension service or a Kubernetes service.

Reply:
> What's ambiguous is to a user the api states it's a service ref which could be an extension service or a Kubernetes service.

I think that's a valid point, disambiguating from a Kubernetes `service` would be helpful.

Reply:
I don't think that this is a problem. Maybe `extensionRef` could be an alternative though.

Reply:
Actually I'm reconsidering that. Maybe naming the field extensionRef binds it to the referent kind a bit too tightly.

Reply:
This reference already uses a layer of indirection by using the ExtensionServiceReference, which, as written, specifically allows multiple Kinds. 

That said, I think `extensionRef` is okay, because the actual type of the object is `ExtensionServiceReference`, I think that is clearer.